arrow/src/main/java/org/apache/iceberg/arrow/vectorized/GenericArrowVectorAccessorFactory.java

```diff
@@ -166,12 +169,20 @@ public ArrowVectorAccessor<DecimalT, Utf8StringT, ArrayT, ChildVectorT> getVecto
 
   @SuppressWarnings("checkstyle:CyclomaticComplexity")
   private ArrowVectorAccessor<DecimalT, Utf8StringT, ArrayT, ChildVectorT> getPlainVectorAccessor(
-      FieldVector vector) {
+      FieldVector vector, PrimitiveType primitive) {
+    boolean isDecimal =
+        primitive != null && OriginalType.DECIMAL.equals(primitive.getOriginalType());
     if (vector instanceof BitVector) {
       return new BooleanAccessor<>((BitVector) vector);
     } else if (vector instanceof IntVector) {
+      if (isDecimal) {
```

Comment:
can we move `isDecimal` to a private method and only invoke it in the specific if branches? Because these branches are mutually exclusive, we don't need to compute it for all the other cases.

Reply:
Updated